plugins/coverimage.koplugin/main.lua

```diff
@@ -0,0 +1,268 @@
+local Device = require("device")
+
+if not Device.isAndroid() and not Device.isEmulator() then
+    return { disabled = true }
+end
+
+local InfoMessage = require("ui/widget/infomessage")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local lfs = require("libs/libkoreader-lfs")
+local logger = require("logger")
+local util = require("util")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local function pathOk(filename)
+    filename = filename:gsub("/$", "")
+    local path = util.splitFilePathName(filename)
+    return Device:isValidPath(path)
+end
+
+local CoverImage = WidgetContainer:new{
+    name = 'coverimage',
+    is_doc_only = true,
+}
+
+function CoverImage:init()
+    self.cover_image_path = G_reader_settings:readSetting("cover_image_path") or "Cover.png"
+    self.cover_image_fallback_path = G_reader_settings:readSetting("cover_image_fallback_path") or "Cover_fallback.png"
+    self.enabled = G_reader_settings:isTrue("cover_image_enabled")
+    self.fallback = G_reader_settings:isTrue("cover_image_fallback")
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function CoverImage:_enabled()
+    return self.enabled
+end
+
+function CoverImage:_fallback()
+    return self.fallback
+end
+
+function CoverImage:dubiousFallbackImage()
+    UIManager:show(InfoMessage:new{
+        text = T(_("\"%1\" \nis not a valid image file!\nPlease correct fallback image in Cover-Image"), self.cover_image_fallback_path),
+        show_icon = true,
+        timeout = 10,
+    })
+end
+
+function CoverImage:showWrongPath( path )
+    UIManager:show(InfoMessage:new{
+        text = T(_("Path of \"%1\" is not accessible.\nPlease correct it."), path),
+        show_icon = true,
+    })
+end
+
+function CoverImage:cleanUpImage()
+    if self.cover_image_fallback_path == "" or not self.fallback then
+        os.remove(self.cover_image_path)
+    elseif lfs.attributes(self.cover_image_fallback_path, "mode") ~= "file" then
+        self:dubiousFallbackImage()
+        os.remove(self.cover_image_path)
+    else
+        os.execute("cp " .. self.cover_image_fallback_path .. " " .. self.cover_image_path)
```

Comment:
Err, speaking of, what's wrong with https://github.com/koreader/koreader-base/blob/master/ffi/util.lua#L197?

We shouldn't *ever* need to shell out to cp?

Reply:
What's wrong with it is that I didn't even know/remember we had it :) (I looked around frontend util.removeFile for a cp equivalent - forgot to look in base ffi/util.lua)

Reply:
I'll admit I kinda expected it in frontend, too. But then it's around path-related shenanigans that do belong in ffi because of win32 shenanigans, so, eeeh.

Reply:
> Err, speaking of, what's wrong with

Argh, I was semi-thinking "I guess we really need one of those" but figured I already had so much to say. Since we already have it that's a different story! ;-)

Reply:
Ok, it uses `ffiutil.copyFile` now.